test/e2e/coderef/code_reference_test.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2025 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coderef
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+
+	featureclient "github.com/bucketeer-io/bucketeer/pkg/feature/client"
+	rpcclient "github.com/bucketeer-io/bucketeer/pkg/rpc/client"
+	coderefproto "github.com/bucketeer-io/bucketeer/proto/coderef"
+	featureproto "github.com/bucketeer-io/bucketeer/proto/feature"
+)
+
+const (
+	timeout = 30 * time.Second
```

Comment:
Due to GitHub Actions instabilities, let's increase it to 1 minute.
```suggestion
	timeout = 60 * time.Second
```